### PR TITLE
(DOCSP-24458): Convert existing SwiftUI examples to new test app and docs structure

### PR DESCRIPTION
**Pull request by dacharyc**
### Jira

- https://jira.mongodb.org/browse/DOCSP-24458

### Staged Changes (Requires MongoDB Corp SSO)

- [SwiftUI](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-24458/sdk/swift/swiftui/): New section. Examples are mostly based on existing examples and text from what is currently the [Use Realm Database with SwiftUI page](https://www.mongodb.com/docs/realm/sdk/swift/swiftui-guide/), but they use the new "SwiftUICatalog" test app and SwiftUICatalogUITests.

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/swift/swiftui/write.txt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+Update an ObservedResults Collection
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this example, we create a two-way binding with an 
+:swift-sdk:`ObservedResults <Structs/ObservedResults.html>` collection.
+The ``$dogs.remove`` in the ``onDelete`` and the ``$dogs.append`` in the 
+``addDogButton`` write directly to the ``@ObservedResults`` collection.
+While a regular :ref:`Realm Results collection <ios-results-collections>`
+is immutable, ObservedResults is a mutable collection that allows you to 
+implicitly perform writes using the two-way binding.
```
Comment (cbush) [inline source/sdk/swift/swiftui/write.txt:47]: I don't know about "implicitly perform writes". I definitely could be wrong, but I thought it's more like "allows you to write by implicitly opening a write transaction".

File: source/sdk/swift/swiftui/filter-data.txt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+Search a Realm Collection
+-------------------------
+
+.. versionadded:: 10.19.0
+
+The Realm Swift SDK allows you to extend :apple:`.searchable 
+<swiftui/form/searchable(text:placement:prompt:)-6royb>`. When you use 
+:swift-sdk:`ObservedResults <Structs/ObservedResults.html>` to query a realm,
+you can specify collection and keypath in the result set to denote it is
```
Comment (cbush) [inline source/sdk/swift/swiftui/filter-data.txt:21]: What are collection and keypath supposed to be?

Comment (cbush) [inline source/sdk/swift/swiftui/filter-data.txt:21]: Ultra-nitpick: For some reason the word "denote" really sticks out to me as rare:
```suggestion
you can specify collection and keypath in the result set to mark it as
```

File: source/sdk/swift/swiftui/swiftui-previews.txt [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+
+.. code-block:: swift
+
+   static var previewRealm: Realm {
+      var realm: Realm
+      let identifier = "previewRealm"
+      let config = Realm.Configuration(inMemoryIdentifier: identifier)
+      do {
+         realm = try Realm(configuration: config)
+         ... Add data to realm
```
Comment (cbush) [inline source/sdk/swift/swiftui/swiftui-previews.txt:232]:
```suggestion
         // ... Add data to realm
```

**Review by cbush — APPROVED** [review]
Excellent! Love it! Great work!

Many comments (as this was a big PR) but nothing blocking.

File: source/sdk/swift/swiftui/pass-realm-data-between-views.txt [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+If you are using :ref:`Partition-Based Sync <partition-based-sync>`, you 
+can use environment injection to pass the ``.partitionValue`` environment 
+value. Inject this into a view where you perform the ``@AsyncOpen`` or 
+``@AutoOpen``:
+
+.. literalinclude:: /examples/generated/swiftui/Authenticate.snippet.partition-value-environment-object.swift
+   :language: swift
+
+Then, when you use the property wrapper to open a synced realm, leave
+the ``partitionValue`` an empty string. The property wrapper populates the
```
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/sdk/swift/swiftui/pass-realm-data-between-views.txt:57]: Working through review comments, the short answer to this one is: "I don't remember." I know we had a big discussion about it (last year?) when it came out, but I don't recall all the details anymore. I'll investigate. 👍

**Pull request merged**